apps/omg_eth/test/fixtures.exs

```diff
@@ -43,47 +44,37 @@ defmodule OMG.Eth.Fixtures do
 
     contracts = SnapshotContracts.parse_contracts()
 
-    contract = %{
-      authority_addr: Encoding.from_hex(contracts["AUTHORITY_ADDRESS"]),
-      contract_addr: %{
-        erc20_vault: Encoding.from_hex(contracts["CONTRACT_ADDRESS_ERC20_VAULT"]),
-        eth_vault: Encoding.from_hex(contracts["CONTRACT_ADDRESS_ETH_VAULT"]),
-        payment_exit_game: Encoding.from_hex(contracts["CONTRACT_ADDRESS_PAYMENT_EXIT_GAME"]),
-        plasma_framework: Encoding.from_hex(contracts["CONTRACT_ADDRESS_PLASMA_FRAMEWORK"])
-      },
-      txhash_contract: Encoding.from_hex(contracts["TXHASH_CONTRACT"])
-    }
-
     {:ok, true} =
       Ethereumex.HttpClient.request("personal_unlockAccount", ["0x6de4b3b9c28e9c3e84c2b2d3a875c947a84de68d", "", 0], [])
 
-    add_exit_queue = RootChainHelper.add_exit_queue(@test_eth_vault_id, @eth, contract.contract_addr)
+    :ok = System.put_env("ETHEREUM_NETWORK", "LOCALCHAIN")
+    :ok = System.put_env("TXHASH_CONTRACT", contracts["TXHASH_CONTRACT"])
+    :ok = System.put_env("AUTHORITY_ADDRESS", contracts["AUTHORITY_ADDRESS"])
+    :ok = System.put_env("CONTRACT_ADDRESS_PLASMA_FRAMEWORK", contracts["CONTRACT_ADDRESS_PLASMA_FRAMEWORK"])
+    SetContract.init([])
+
+    add_exit_queue = RootChainHelper.add_exit_queue(@test_eth_vault_id, "0x0000000000000000000000000000000000000000")
 
     {:ok, %{"status" => "0x1"}} = Support.DevHelper.transact_sync!(add_exit_queue)
 
-    contract
+    :ok
   end
 
   deffixture token(root_chain_contract_config) do
     :ok = root_chain_contract_config
     contracts = SnapshotContracts.parse_contracts()
-
-    token_addr = Encoding.from_hex(contracts["CONTRACT_ERC20_MINTABLE"])
+    token_addr = contracts["CONTRACT_ERC20_MINTABLE"]
 
     # ensuring that the root chain contract handles token_addr
-    {:ok, false} = RootChainHelper.has_exit_queue(@test_erc20_vault_id, token_addr)
+    {:ok, false} = has_exit_queue(@test_erc20_vault_id, token_addr)
     {:ok, _} = DevHelper.transact_sync!(RootChainHelper.add_exit_queue(@test_erc20_vault_id, token_addr))
-    {:ok, true} = RootChainHelper.has_exit_queue(@test_erc20_vault_id, token_addr)
+    {:ok, true} = has_exit_queue(@test_erc20_vault_id, token_addr)
 
     token_addr
   end
 
   deffixture root_chain_contract_config(contract) do
-    contract_addr = contract_map_to_hex(contract.contract_addr)
-    Application.put_env(:omg_eth, :contract_addr, contract_addr, persistent: true)
-    Application.put_env(:omg_eth, :authority_addr, Encoding.to_hex(contract.authority_addr), persistent: true)
-    Application.put_env(:omg_eth, :txhash_contract, Encoding.to_hex(contract.txhash_contract), persistent: true)
-
+    _ = contract
```

Comment:
yeah, otherwise the contract variable is `unused`, but the way these fixtures work is that they're chained together (and the order) based on the arguments between them. it's funny I tell you.

Reply:
Ah i see.......................................